AutoGear.lua

```diff
@@ -2413,14 +2414,15 @@ AutoGearFrame:SetScript("OnEvent", function (this, event, arg1, arg2, arg3, arg4
 			local soldSomething = nil
 			local totalSellValue = 0
 			for i = 0, NUM_BAG_SLOTS do
-				local slotMax = GetContainerNumSlots(i)
+				local slotMax = ((C_Container and C_Container.GetContainerNumSlots) or GetContainerNumSlots)(i)
 				for j = 0, slotMax do
-					local _, count, locked, quality, _, _, link = GetContainerItemInfo(i, j)
+					local _, count, locked, quality, _, _, link = ((C_Container and C_Container.GetContainerItemInfo) or GetContainerItemInfo)(i, j)
 					if (link) then
 						local name = select(3, string.find(link, "^.*%[(.*)%].*$"))
 						if (string.find(link,"|cff9d9d9d") and not locked and not AutoGearIsQuestItem(i,j)) then
 							totalSellValue = totalSellValue + select(11, GetItemInfo(link)) * count
-							PickupContainerItem(i, j)
+							-- For some reason I could not identify, this does not work with the cascade logic used elsewhere for cross compatibility
+							C_Container.PickupContainerItem(i, j)
```

Comment:
You could write it this way, which might actually be my preference for this kind of thing:

```Lua
if C_Container and C_Container.PickupContainerItem then
	C_Container.PickupContainerItem(i, j)
elseif PickupContainerItem then
	PickupContainerItem(i, j)
else
	AutoGearPrint("AutoGear: Can't pick up an item because function PickupContainerItem does not exist in this version of the WoW addon API ("..tostring(TOC_VERSION_CURRENT)..").  Returning.",0)
	return
end
```

This way, you're not relying on the return value of the ternary, and simply relying on the true or nil of the if conditional before executing the exact function you intend.  I'm not sure if there's a real difference, but it seems clearer to me and allows for additional logic like an error message and early return.

Reply:
I can make that change